### PR TITLE
fix(tab): transitions

### LINKED ISSUE #504: Tabs not fading
It is stated in the docs that fade is enabled by default but the tabs don't fade. This can be seen in the corresponding [docs page](https://bootstrap-vue.github.io/docs/components/tabs) itself.

Maybe something changed in bootstrap that broke the fading? (Is something like that even possible?)

### PR DESCRIPTION
**Pull request by mosinve**
Make it transition again
Attempt to fix #504

### CODE REVIEW

File: lib/components/tab.vue [hunk 1]
```diff
@@ -7,8 +7,7 @@
                    :aria-hidden="localActive ? 'false' : 'true'"
                    :aria-expanded="localActive ? 'true' : 'false'"
                    :aria-lablelledby="controlledBy || null"
-                   v-if="localActive || !lazy"
-                   v-show="localActive || lazy"
+                   v-show="localActive"
```
Comment (tmorehouse) [inline lib/components/tab.vue]: what if this was:

```html
v-if="localActive && lazy"
v-show="localActive"
```
Would that make a difference?

Some people may not want any child components of a tab rendered if not visible.

Comment (mosinve) [inline lib/components/tab.vue]: if there's a v-show, then element will be rendered

Comment (mosinve) [inline lib/components/tab.vue]: but if only v-if stay... transitions are buggy...

Comment (mosinve) [inline lib/components/tab.vue]: then .. judgement is simple- v-show is the only case when this component works just like original BS4

Comment (tmorehouse) [inline lib/components/tab.vue]: But removing the lazy part will always render with v-show (but just have `display:none;` set), so the content of the tabs will always be in the dom (and potentially active) even though not visible, which may not be the desired state for some users

Comment (tmorehouse) [inline lib/components/tab.vue]: Maybe if we move the lazy part to the tab content slot?  So the tab itself is always present 9but hiding with v-show), but the content inside can be taken out of DOM via a v-if?

Comment (tmorehouse) [inline lib/components/tab.vue]: What aaaaabout this:

````
                   v-show="localActive"
                   ref="panel"
        >
             <template v-if="localActive || !lazy">
                 <slot></slot>
             </template>
</component>
```

Comment (mosinve) [inline lib/components/tab.vue]: Check updated file... seems to work, at least at playground

Comment (mosinve) [inline lib/components/tab.vue]: the catch is we must spread active and show classes in time...

**Pull request merged**